apisix/plugins/ai.lua

```diff
@@ -100,33 +108,120 @@ local function gen_get_cache_key_func(route_flags)
 end
 
 
+local function ai_upstream()
+    core.log.info("enable sample upstream")
+end
+
+
+local pool_opt = { pool_size = 320 }
+local function ai_balancer_run(route)
+    local server = route.value.upstream.nodes[1]
+    if enable_keepalive then
+        local ok, err = balancer.set_current_peer(server.host, server.port or 80, pool_opt)
+        if not ok then
+            core.log.error("failed to set server peer [", server.host, ":",
+                           server.port, "] err: ", err)
+            return ok, err
+        end
+        balancer.enable_keepalive(60, 1000)
+    else
+        balancer.set_current_peer(server.host, server.port or 80)
+    end
```

Comment:
We can move:
```
if enable_keepalive then
```
to here?
Like:
```
if enable_keepalive then
    balancer.enable_keepalive(60, 1000)
```

Reply:
maybe not, follow: https://github.com/apache/apisix/blob/ecdc209523f011b5fb4a0fee3a8e867da9c53fb5/apisix/balancer.lua#L291-L338

if `enable_keepalive`, would add `pool_opt` as the third argument to `balancer.set_current_peer`.

Do we fix the call to `set_current_peer` here and in `balancer.lua`, pass `pool_opt` even if there is disable `enable_keepalive`, otherwise it will all become non-persistent connection.